test/e2e/e2e_test.go

```diff
@@ -1,474 +0,0 @@
-	<-clientCtx.Done()
-
-	log().Infof("Took %v to process", end.Sub(start))
-	_ = s.Shutdown()
-
-	// Should have been grabbed by queue processor
-	q := s.QueuesForE2EOnly()
-	assert.Equal(t, 0, q.RecvQ("default").Len())
-
-	// All applications should have been created by now on the server
-	apps, err := appC.ArgoprojV1alpha1().Applications("default").List(context.Background(), v1.ListOptions{})
-	assert.NoError(t, err)
-	assert.Len(t, apps.Items, 10)
-}
-
-func Test_AgentServer(t *testing.T) {
-	app := &v1alpha1.Application{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      "testapp",
-			Namespace: "client",
-		},
-	}
-	sctx, scancel := context.WithTimeout(context.Background(), 20*time.Second)
-	actx, acancel := context.WithTimeout(context.Background(), 20*time.Second)
-	fakeAppcServer := fakekube.NewKubernetesFakeClientWithApps()
-	am := auth.NewMethods()
-	up := userpass.NewUserPassAuthentication("")
-	err := am.RegisterMethod("userpass", up)
-	require.NoError(t, err)
-	up.UpsertUser("client", "insecure")
-	s, err := principal.NewServer(sctx, fakeAppcServer, "server",
-		principal.WithGRPC(true),
-		principal.WithListenerPort(0),
-		principal.WithServerName("control-plane"),
-		principal.WithGeneratedTLS("control-plane"),
-		principal.WithAuthMethods(am),
-		principal.WithNamespaces("client"),
-		principal.WithGeneratedTokenSigningKey(),
-	)
-	require.NoError(t, err)
-	require.NotNil(t, s)
-	errch := make(chan error)
-	_ = s.Start(sctx, errch)
-	defer scancel()
-	defer acancel()
-
-	remote, err := client.NewRemote(s.ListenerForE2EOnly().Host(), s.ListenerForE2EOnly().Port(),
-		client.WithInsecureSkipTLSVerify(),
-		client.WithAuth("userpass", auth.Credentials{userpass.ClientIDField: "client", userpass.ClientSecretField: "insecure"}),
-	)
-	require.NoError(t, err)
-	fakeKubeClient := fakekube.NewKubernetesFakeClientWithApps()
-	a, err := agent.NewAgent(actx, fakeKubeClient, "client",
-		agent.WithRemote(remote),
-		agent.WithMode(types.AgentModeManaged.String()),
-	)
-	require.NotNil(t, a)
-	require.NoError(t, err)
-	_ = a.Start(actx)
-	for !a.IsConnected() {
-		log().Infof("waiting to connect")
-		time.Sleep(100 * time.Millisecond)
-	}
-	log().Infof("Creating application")
-	_, err = fakeAppcServer.ApplicationsClientset.ArgoprojV1alpha1().Applications("client").Create(sctx, app, v1.CreateOptions{})
-	require.NoError(t, err)
-	for i := 0; i < 5; i += 1 {
-		app, err = fakeAppcServer.ApplicationsClientset.ArgoprojV1alpha1().Applications("client").Get(actx, "testapp", v1.GetOptions{})
-		if err == nil {
-			break
-		}
-		time.Sleep(1 * time.Second)
-	}
-	require.NoError(t, err)
-	require.NotNil(t, app)
-	app.Spec.Project = "hulahup"
-	time.Sleep(1 * time.Second)
-	_, err = fakeAppcServer.ApplicationsClientset.ArgoprojV1alpha1().Applications("client").Update(actx, app, v1.UpdateOptions{})
-	require.NoError(t, err)
-
-	<-sctx.Done()
-	<-actx.Done()
-	err = s.Shutdown()
-	require.NoError(t, err)
-}
-
-func Test_WithHTTP1WebSocket(t *testing.T) {
```

Comment:
Should we keep this test since we support HTTP 1 downgrade? I can port it to the new e2e framework in a different PR.